### PR TITLE
[Merged by Bors] - Fix comment typos

### PR DESCRIPTION
**Pull request by bilsen**
Fix some typos in system_param.rs

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/system/system_param.rs [hunk 1]
```diff
@@ -444,7 +444,7 @@ impl<'w, 's, T: Component> SystemParamFetch<'w, 's> for ResMutState<T> {
     }
 }
 
-/// The [`SystemParamState`] of `Option<ResMut<T>>`.
+/// The [`SystemParamState`] of [`Option<ResMut<T>>`].
```
Comment (mockersf) [inline crates/bevy_ecs/src/system/system_param.rs:447]: This make it link to `Option` in the rust doc... Could you also add a link to `ResMut<T>`?

Comment (bilsen) [reply crates/bevy_ecs/src/system/system_param.rs:448]: Done

**Review by mockersf — APPROVED** [review]

File: crates/bevy_ecs/src/system/system_param.rs (hunk 1 above)
Comment (mockersf) [reply crates/bevy_ecs/src/system/system_param.rs:448]: thanks!

**Pull request closed without merging**